# Part 1: data preprocessing and restructuring

In [1]:
import pandas as pd
import time 

#read the original file
#use chunksize to reduce the loading time

start = time.time()
df = pd.read_csv('Assignment1_data.csv',chunksize=1000000)
# print(df.to_string())
end = time.time()
pd_df = pd.concat(df)

print("time taken to read csv with chunks:",(end-start),"sec")

time taken to read csv with chunks: 0.0041048526763916016 sec


In [2]:
print("shape:",pd_df.shape)
print("rows:", pd_df.shape[0])
print("columns:",pd_df.shape[1])

shape: (3574594, 10)
rows: 3574594
columns: 10


In [3]:
print(pd_df.head())

        ID               Date_Time  Year     Month  Mdate     Day  Time  \
0  2887628  11/01/2019 05:00:00 PM  2019  November      1  Friday    17   
1  2887629  11/01/2019 05:00:00 PM  2019  November      1  Friday    17   
2  2887630  11/01/2019 05:00:00 PM  2019  November      1  Friday    17   
3  2887631  11/01/2019 05:00:00 PM  2019  November      1  Friday    17   
4  2887632  11/01/2019 05:00:00 PM  2019  November      1  Friday    17   

   Sensor_ID                   Sensor_Name Hourly_Counts  
0         34          Flinders St-Spark La           300  
1         39                  Alfred Place           604  
2         37               Lygon St (East)           216  
3         40  Lonsdale St-Spring St (West)           627  
4         36               Queen St (West)           774  


In [4]:
df_nodup = pd_df.drop_duplicates()

In [6]:
#write the date_info file 
start = time.time()
df_nodup[['ID','Date_Time','Day']].to_csv('date_info2.csv',header=False,index=False)
end = time.time()
print("time taken to write date_info:",(end-start),"sec")
print("shape:",df_nodup.shape)
print("rows:", df_nodup.shape[0])
print("columns:",df_nodup.shape[1])

time taken to write date_info: 6.136405944824219 sec
shape: (3574594, 10)
rows: 3574594
columns: 10


In [7]:
#write the sensor_info file 
start = time.time()
df_nodup[['Sensor_ID','Sensor_Name']].to_csv('sensor_info2.csv',header=False,index=False)
end = time.time()
print("time taken to write sensor_info:",(end-start),"sec")
print("shape:",df_nodup.shape)
print("rows:", df_nodup.shape[0])
print("columns:",df_nodup.shape[1])

time taken to write sensor_info: 4.703268051147461 sec
shape: (3574594, 10)
rows: 3574594
columns: 10


In [8]:
#write the sensor_info file 
start = time.time()
df_nodup[['ID','Sensor_ID','Hourly_Counts']].to_csv('count_info2.csv',header=False,index=False)
end = time.time()
print("time taken to write count_info:",(end-start),"sec")
print("shape:",df_nodup.shape)
print("rows:", df_nodup.shape[0])
print("columns:",df_nodup.shape[1])

time taken to write count_info: 4.356667995452881 sec
shape: (3574594, 10)
rows: 3574594
columns: 10


time taken to read csv with chunks: 0.0041048526763916016 sec
time taken to write date_info: 6.136405944824219 sec
time taken to write sensor_info: 4.703268051147461 sec
time taken to write count_info: 4.356667995452881 sec

# Part 2: scripts for loading .csv file to derby database